src/StackExchange.Redis/ConnectionMultiplexer.cs

```diff
@@ -876,13 +882,20 @@ private static async Task<ConnectionMultiplexer> ConnectImplAsync(object configu
         /// </summary>
         /// <param name="configuration">The configuration options to use for this multiplexer.</param>
         /// <param name="log">The <see cref="TextWriter"/> to log to.</param>
-        public static Task<ConnectionMultiplexer> ConnectAsync(ConfigurationOptions configuration, TextWriter log = null)
+        public static async Task<ConnectionMultiplexer> ConnectAsync(ConfigurationOptions configuration, TextWriter log = null)
         {
             SocketConnection.AssertDependencies();
-            return ConnectImplAsync(configuration, log);
+
+            bool sentinel = !String.IsNullOrEmpty(configuration.ServiceName);
+
+            if (!sentinel)
+                return await ConnectImplAsync(PrepareConfig(configuration), log).ForAwait();
+
+            var conn = await ConnectImplAsync(PrepareConfig(configuration, true), log).ForAwait();
+            return conn.GetSentinelMasterConnection(PrepareConfig(configuration), log);
```

Comment:
I'm turn here, because this path _isn't_ async, but I don't really fancy adding a whole `GetSentinelMasterConnectionAsync` path either - @mgravell thoughts?

Reply:
Yeah, that’s the same thing I was thinking. Didn’t seem worth having a whole new async copy of that method.